flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -66,31 +80,48 @@
 	//  Consumer properties
 	// ------------------------------------------------------------------------
 
-	/** The names of the Kinesis streams that we will be consuming from */
+	/**
+	 * The names of the Kinesis streams that we will be consuming from
+	 */
 	private final List<String> streams;
 
-	/** Properties to parametrize settings such as AWS service region, initial position in stream,
-	 * shard list retrieval behaviours, etc */
+	/**
+	 * Properties to parametrize settings such as AWS service region, initial position in stream,
+	 * shard list retrieval behaviours, etc
+	 */
 	private final Properties configProps;
 
-	/** User supplied deseriliazation schema to convert Kinesis byte messages to Flink objects */
+	/**
+	 * User supplied deseriliazation schema to convert Kinesis byte messages to Flink objects
+	 */
 	private final KinesisDeserializationSchema<T> deserializer;
 
 	// ------------------------------------------------------------------------
 	//  Runtime state
 	// ------------------------------------------------------------------------
 
-	/** Per-task fetcher for Kinesis data records, where each fetcher pulls data from one or more Kinesis shards */
+	/**
+	 * Per-task fetcher for Kinesis data records, where each fetcher pulls data from one or more Kinesis shards
+	 */
 	private transient KinesisDataFetcher<T> fetcher;
 
-	/** The sequence numbers in the last state snapshot of this subtask */
+	/**
+	 * The sequence numbers in the last state snapshot of this subtask
+	 */
 	private transient HashMap<KinesisStreamShard, SequenceNumber> lastStateSnapshot;
 
-	/** The sequence numbers to restore to upon restore from failure */
+	/**
+	 * The sequence numbers to restore to upon restore from failure
+	 */
 	private transient HashMap<KinesisStreamShard, SequenceNumber> sequenceNumsToRestore;
 
 	private volatile boolean running = true;
 
+	// ------------------------------------------------------------------------
+	// State for Checkpoint
+	// ------------------------------------------------------------------------
+
+	private transient ListState<Tuple2<KinesisStreamShard, SequenceNumber>> offsetsStateForCheckpoint;
```

Comment:
"offset" is the Kafka term.
I would try to rename this to use "sequence number" instead (or a likewise abbreviation).